File: OLSource1/v1/VS_officedev/codesnippet/CSharp/how-to--cache-data-in-a-password-protected-document_1.cs
        [CachedAttribute]
        public string CachedString = "This string is cached in the document.";

        private Word.WdProtectionType protectionTypeValue;

        protected override void UnprotectDocument()
        {
            if (this.ProtectionType != Word.WdProtectionType.wdNoProtection)
            {
                protectionTypeValue = this.ProtectionType;
                this.Unprotect(ref securelyStoredPassword);
            }
        }

        protected override void ProtectDocument()
        {
            this.Protect(protectionTypeValue, ref missing,
                ref securelyStoredPassword, ref missing, ref missing);
        }

File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/how-to--cache-data-in-a-password-protected-document_1.vb
    <CachedAttribute()> _
    Public CachedString As String = "This string is cached in the document."

    Private protectionTypeValue As Word.WdProtectionType

    Protected Overrides Sub UnprotectDocument()
        If Me.ProtectionType <> Word.WdProtectionType.wdNoProtection Then
            protectionTypeValue = Me.ProtectionType
            Me.Unprotect(securelyStoredPassword)
        End If
    End Sub

    Protected Overrides Sub ProtectDocument()
        Me.Protect(protectionTypeValue, password:=securelyStoredPassword)
    End Sub